### PR TITLE
Add test plans - alt

### PR DESCRIPTION
**Pull request by dfeinzimer**
After talking briefly with Zach we decided this format makes a bit more sense.

- Renames the `UI Test Runner` project to `Test Runner` to match Swift.
- Adds 2 test plans to the `Test Runner` project, 1 for unit tests and 1 for UI tests.

Ref #294

### CODE REVIEW

**Review by saratchandrakarumuri — DISMISSED** [review]
Test plans are working as intended from both Xcode and command line

File: Test Runner/Test Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -77,6 +79,8 @@
 		7552A7472A573CBB0023DA5A = {
 			isa = PBXGroup;
 			children = (
+				8AE5AE902B85498900FE3759 /* UI Tests.xctestplan */,
+				8AE5AE8F2B85495400FE3759 /* Unit Tests.xctestplan */,
```
Comment (zkline101) [inline Test Runner/Test Runner.xcodeproj/project.pbxproj:83]: These can go under a `Test Plans` folder

**Review by zkline101 — COMMENTED** [review]
I see some more folders/files that are under the old name. Some of the files `UI Test Runner` need updates too.
Screenshot 2024-02-21 at 8 48 03 AM

Thanks for these changes, I know renaming an Xcode project is a pain.

File: Test Runner/Test Runner.xcodeproj/project.pbxproj (hunk 1 above)
Comment (dfeinzimer) [reply Test Runner/Test Runner.xcodeproj/project.pbxproj:83]: Hmm not sure what to change as those two files are in a Test Plans folder in the top level directory.

Are these just references?

**Comment by dfeinzimer** [comment]
> I see some more folders/files that are under the old name. Some of the files UI Test Runner need updates too.

Resolved. This is ready for review once again. 

@saratchandrakarumuri note that `UI Test Runner/UI Test Runner/UITestRunner.swift` has become `Test Runner/Test Runner/TestRunnerApp.swift`.

**Review by zkline101 — COMMENTED** [review]
We should change the bundle ID too, right? Or will that affect other things as well?
Screenshot 2024-02-21 at 1 33 52 PM

**Comment by dfeinzimer** [comment]
> We should change the bundle ID too, right? Or will that affect other things as well?

Checking on that

**Comment by dfeinzimer** [comment]
This is ready for review again.

Bundle IDs for the Test Runner have been renamed and all references to "UI Test Runner" or "ui-test-runner" should be gone from the the project file now.

**Review by zkline101 — APPROVED** [review]

**Review by saratchandrakarumuri — APPROVED** [review]

**Pull request merged**